docker/dockerfile_generator.sh

```diff
@@ -0,0 +1,309 @@
+# distributed under the License is distributed on an \"AS IS\" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+FROM ubuntu:18.04
+
+LABEL maintainer=\"AdoptOpenJDK <adoption-discuss@openjdk.java.net>\"
+  " >> $DFPATH
+}
+
+# Put in apt packages required for building a JDK
+printAptPackages() {
+  [ ${COMMENTS} == true ] && echo "
+# Install required OS tools
+# dirmngr, gpg-agent & coreutils are all required for the apt-add repository command" >> $DFPATH
+
+  echo " 
+RUN apt-get update \\
+  && apt-get install -qq -u --no-install-recommends \\
+    software-properties-common \\
+    dirmngr \\
+    gpg-agent \\
+    coreutils \\
+  && apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 0x219BD9C9 \\
+  && add-apt-repository 'deb http://repos.azulsystems.com/ubuntu stable main' \\
+  && apt-get update \\
+  && apt-get -y upgrade \\
+  && apt-get install -qq -y --no-install-recommends \\
+    autoconf \\
+    cpio \\
+    curl \\
+    file \\
+    git \\
+    libasound2-dev \\
+    libcups2-dev \\
+    libelf-dev \\
+    libfontconfig1-dev \\
+    libfreetype6-dev \\
+    libx11-dev \\
+    libxext-dev \\
+    libxrandr-dev \\
+    libxrender-dev \\
+    libxt-dev \\
+    libxtst-dev \\
+    make \\
+    ssh \\
+    systemtap-sdt-dev \\
+    unzip \\
+    wget \\
+    zip \\" >> $DFPATH
+
+  if [ ${OPENJ9} = true ]; then
+    echo "    libdwarf-dev \\
+    libnuma-dev \\
+    nasm \\
+    pkg-config \\" >> $DFPATH
+  else 
+    echo "    ccache \\
+    g++ \\
+    gcc \\" >> $DFPATH
+  fi
+  
+  # JDK8 uses zulu-7 as it's bootJDK
+  [ ${JDK} == 8 ] && echo "    zulu-7 \\" >> $DFPATH
+
+  echo "  && rm -rf /var/lib/apt/lists/*" >> $DFPATH
+}
+
+printCreateFolder() {
+  echo "
+RUN mkdir -p /openjdk/target
+RUN mkdir -p /openjdk/build" >> $DFPATH
+}
+
+printgcc() {
+  [ ${COMMENTS} == true ] && echo "
+# Make sure build uses GCC 7.3
+# Create links for GCC to access the C library and gcc,g++" >> $DFPATH
+
+  echo "
+RUN cd /usr/local \\
+  && wget -O gcc-7.tar.xz "https://ci.adoptopenjdk.net/userContent/gcc/gcc730+ccache.x86_64.tar.xz" \\
+  && tar -xJf gcc-7.tar.xz \\
+  && rm -rf gcc-7.tar.xz" >> $DFPATH
+
+  echo "
+RUN ln -s /usr/lib/x86_64-linux-gnu /usr/lib64 \\
+  && ln -s /usr/include/x86_64-linux-gnu/* /usr/local/gcc/include \\
+  && ln -s /usr/local/gcc/bin/g++-7.3 /usr/bin/g++ \\
+  && ln -s /usr/local/gcc/bin/gcc-7.3 /usr/bin/gcc \\
+  && ln -s /usr/local/gcc/bin/ccache /usr/local/bin/ccache" >> $DFPATH
+}
+
+printDockerJDKs() {
+  if [ ${JDK} != 8 ]; then
+    [ ${COMMENTS} == true ] && echo "
+    # Extract JDK$((JDK-1)) to use as a boot jdk" >> $DFPATH
+    printJDK $((JDK-1))
+    echo "RUN ln -sf /usr/lib/jvm/jdk$((JDK-1))/bin/java /usr/bin/java" >> $DFPATH
+    echo "RUN ln -sf /usr/lib/jvm/jdk$((JDK-1))/bin/javac /usr/bin/javac" >> $DFPATH
+  fi  
+  if [ ${JDK} != 9 ]; then
```

Comment:
!= to 9 - is there something specific about 9?

Reply:
Yep - Any dockerfile generated is getting JDK8 to run Gradle anyway. If the JDK to build is 9, it will have already got JDK8 at this point and won't need to again :-)